### ISSUE TITLE
domain1 NotEmptyValidationRule

### ISSUE THREAD

**Issue by hazemsaleh** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#1041]
hej 
everything works great but when i use common fields like textbox name and password with vaildation its an error 
<TextBox
  x:Name="NameTextBox"
  materialDesign:HintAssist.Hint="Name">
  <TextBox.Text>
    <Binding
      Path="Name"
      UpdateSourceTrigger="PropertyChanged">
      <Binding.ValidationRules>
        <domain1:NotEmptyValidationRule
          ValidatesOnTargetUpdated="True" xmlns:domain1="clr-namespace:MaterialDesignDemo.Domain" />
      </Binding.ValidationRules>
    </Binding>
  </TextBox.Text>
</TextBox>
vs cannot recognize domain1:NotEmptyValidationRule
any idea why

**Comment by Keboo** [comment]
@hazemsaleh This is because the [NotEmptyValidation](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/blob/master/MainDemo.Wpf/Domain/NotEmptyValidationRule.cs) is part of the demo application and not part of the Material Design in XAML library. If you want to use it you can copy the code from the demo app into your own application. 

If you want more examples of doing data validation you can find them [here](https://github.com/Keboo/MaterialDesignInXaml.Examples/tree/master/MVVM/MVVM.Validation).

**Comment by yanzzz888** [comment]
### Step 1.Add this to you code-behind window.cs
```public class NotEmptyValidationRule : ValidationRule
    {
        public override ValidationResult Validate(object value, CultureInfo cultureInfo)
        {
            return string.IsNullOrWhiteSpace((value ?? "").ToString())
                ? new ValidationResult(false, "Field is required.")
                : ValidationResult.ValidResult;
        }
    }
```

### Step 2.Add this to you  window.xaml



```
            <TextBox.Text>
                    <Binding Path="Name"  UpdateSourceTrigger="PropertyChanged">
                        <Binding.ValidationRules>
                            <local:NotEmptyValidationRule    ValidatesOnTargetUpdated="True"  />
                        </Binding.ValidationRules>
                    </Binding>
             </TextBox.Text>

```
### Step 3.It's working,,,,,,,,,,,,,,,,,,,

**Comment by stefanpolishchuk** [comment]
it not works!

**Comment by karisDev** [comment]
> ### Step 2.Add this to you window.xaml
> 
> ```
>             <TextBox.Text>
>                     <Binding Path="Name"  UpdateSourceTrigger="PropertyChanged">
>                         <Binding.ValidationRules>
>                             <local:NotEmptyValidationRule    ValidatesOnTargetUpdated="True"  />
>                         </Binding.ValidationRules>
>                     </Binding>
>              </TextBox.Text>
> ```

In this code. NotEmptyValidationRule is called only once at the beginning, but will not be called when the field is changed. Must be a problem with:
>ValidatesOnTargetUpdated="True"

or
>UpdateSourceTrigger="PropertyChanged"

I tried to fix it, but my knowledge of XAML is not good enough...

**Comment by CuberLuo** [comment]
@karisDev I also meet that problem.
NotEmptyValidationRule is called only once at the beginning, but will not be called when the field is changed.
